tools/modules.php

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env php
+<?php
+/**
+ * The modules.php script a database and automatically adds any new modules from
+ * either the project/modules or loris/modules directory and maintains the
+ * modules table based on the filesystem state.
+ *
+ * If the --add parameter is provided, it will add any new directories that
+ * have a valid \Module descriptor to the modules table.
+ *
+ * If the --remove parameter is provided, it will remove anything in the
+ * modules table that can not be instantiated with Module::factory from
+ * the modules table.
+ *
+ * If the -n flag is provided, it will not actually add/delete from the
+ * table, but only tell you what it would otherwise do.
+ *
+ * PHP Version 7
+ *
+ * @category   Data
+ * @package    Main
+ * @subpackage Data
+ * @author     Dave MacFarlane <driusan@gmail.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+require_once 'generic_includes.php';
+
+$flags = getopt("nh", ['add', 'remove', 'dry-run', 'help']);
+
+
+$dryrun = isset($flags['n']) || isset($flags['dry-run']);
+$help   = isset($flags['h']) || isset($flags['help']);
+
+if($help) {
+    usage();
+    exit(0);
+}
+
+if(!isset($flags['add']) && !isset($flags['remove'])) {
+    usage();
+    fwrite(STDERR, "\nMust specify --add or --remove flag\n");
+    exit(1);
+
+}
+
+$currentModules = $DB->pselectCol(
+    "SELECT Name FROM modules",
+    []
+);
```

Comment:
shouldn't this query exist somewhere like in the module class or something?

moduleController ???

(out of the scope of this PR )

Reply:
the module class has a static getActiveModules, but this needs the query without the Active='Y' part of the the query. I don't think there's anywhere else in LORIS that we would ever want to get the inactive modules.